### ISSUE TITLE
7.4, GroupID - Questions

### ISSUE THREAD

**Issue by MartinaBerger** [SMPTE/st429-2#77]
I'd like to ask questions regarding the GroupID element:

1. Optional or not? 
There is a DCP validation tool in the field with the option to validate SMPTE DCPs against ST 429-2 . It reports an error when the GroupID element is missing in the PKL of SMPTE VF-DCPs. With "VF-DCP" I mean Version File Packages/ Supplemental Packages that are apparently called Asset Packages in terms of 429-2 (if I'm right).
I've never seen the GroupID element in SMPTE VF-DCPs in practice so far. In ST 429-8 the GroupID element in section 5.7 holds the word "optional" in square brackets. However, in ST 429-2 section 7.4.2 it seems to be strictly required. And in RDD 52 I can't find anything about the GroupID element. 
So I'm confused whether this element shall be present in SMPTE VF-DCPs or not. And if so, shall it be present in all SMPTE flavors that are currently in the field (SMPTE A, SMPTE B2.0, SMPTE B2.1)?

2. Usage of the GroupID element 
Some diagrams of relations between different packages might be helpful in section 7.4.2. in order to explain the following two cases clearer:
A. usage of the same GroupID element value 
B. usage of different GroupID element values
On basis of the pure text, it's honestly difficult for me to understand. These diagrams could also explain, which UUID specifically shall be used as the value for the GroupID element in the PKL of an Asset Package. E.g. is it the UUID of the PKL that completes the composition (= pkl:id of the OV-Package/ Full Package that is apparently called Composition Package in terms of ST 429-2)? Does it make a difference when the PKL of the OV-Package contains more than one CPL? 
And how to deal with the GroupID element value in the rare case in D-Cinema, when for the playback of a VF-DCP an OV-DCP _and_ a second VF-DCP is needed?

Thanks!

**Comment by JBoisseau-ByDeluxe** [comment]
(For reference, in the latest 429-2:2023 revision, this section has been moved to 7.5, but still unchanged.)

I share @MartinaBerger concerns.
Considering the GroupID concept has not been implemented (afaik) and the concept is not well enough defined, I suggest to align the Standard to the usage by simply removing the section 7.5.2.

related issue:
https://github.com/SMPTE/st429-8/issues/7

**Comment by SteveLLamb** [comment]
We go with the schema rules on this being optional in all cases. 

IMO the concept of `groupId` should just be removed, it's never been used, and never implemented on server side afaik. The issue is that you need to be aware of any mapping from `groupId` back to the non-`groupId` "base package. Good idea, but there have been other ways this has been accomplished, no need to add bloat here.

**Comment by jhursty** [comment]
I agree that `groupId` should be optional for the reasons given above.

I do not agree with the policy suggestion "We go with the schema rules on this being optional in all cases." Syntactically optional items may be logically required in some circumstances. For example, in clause 9.11 "Hash Element" in this same document, "The Hash element shall be present in an asset when the KeyId element is present (i.e., when the referenced Track File is encrypted)." The CPL schema says it is optional but 429-2 makes it required in some cases.
